docs/DeveloperGuide.md

```diff
@@ -362,6 +312,126 @@ The above activity diagram shows the logic and the path execution when the switc
     * Cons:
         * This introduces the need to type twice in order to view the execution of the command.
 
+### Storage Feature
+
+#### What it is
+After a command is successfully executed, InternHunter automatically saves users' data to JSON files. Moreover, 
+everytime the `GuiSettings` is modified, InternHunter updates the user preferences JSON file. Users can transfer or 
+backup the JSON files manually. The storage component is responsible for both reading and saving the data.
+
+#### Implementation
+InternHunter uses Jackson, a high-performance JSON processor for Java. It can  serialize Java objects into JSON and 
+deserialize JSON into Java objects. InternHunter's model has 4 different types of data: `ApplicationItem`, 
+`CompanyItem`, `InternshipItem`, and `ProfileItem`. They first need to be converted to Jackson-friendly versions of 
+themselves, where each field is a string or another Jackson-friendly object. User preference is saved as a `UserPrefs` 
+object.
+ 
+ [Storage structure diagram](#storage-class-diagram)
+ 
+ * `Storage` handles the storage for all `Item` lists and user preferences.
+ * `UserPrefsStorage` handles the storage for user preferences.
+ * `ItemListStorage` handles the storage for `Item` lists.
+ * `JsonSerializableItemList` represents a Jackson-friendly version of an `Item` list.
+ * `JsonAdaptedItem` represents a Jackson-friendly version of an `Item`.
+ 
+ ![JsonAdaptedItemClassDiagram](images/JsonAdaptedItemClassDiagram.png)
+ 
+ `JsonAdaptedItem` is an abstract class representing Jackson-friendly version of the `Item` class in the model component.
+  It has one method `toModelType()` which convert itself to an `Item` object. There are 4 classes extending 
+  `JsonAdaptedItem`:
+  * `JsonAdaptedApplicationItem` the Jackson-friendly version of `ApplicationItem`.
+  * `JsonAdaptedCompanyItem` the Jackson-friendly version of `CompanyItem`.
+  * `JsonAdaptedInternshipItem` the Jackson-friendly version of `InternshipItem`.
+  * `JsonAdaptedProfileItem` the Jackson-friendly version of `ProfileItem`.
+  
+InternHunter automatically saves user data after every command. The following sequence diagram demonstrates how
+InternHunter does it. Let `commandString` be any valid command string.
+  
+![SavingDataSequenceDiagram](images/SavingDataSequenceDiagram.png)
```

Comment:
Not sure why its not rendering on github? does your image render properly locally?

Reply:
Also I don't think StorageManager is in :Model?

Reply:
nice catch! and it does